search/wildcard_query.go

```diff
@@ -0,0 +1,89 @@
+package search
+
+/*
+#include <ffi_go/defs.h>
+#include <ffi_go/search/wildcard_query.h>
+*/
+import "C"
+
+import (
+	"strings"
+	"unsafe"
+
+	"github.com/y-scope/clp-ffi-go/ffi"
+)
+
+// A CLP wildcard query containing a query string and a bool for whether the
+// query is case sensitive or not. The fields must be accessed through getters
+// to ensure that the query string remains clean/safe after creation by
+// NewWildcardQuery.
+// (Copied from clp/components/core/src/string_utils.hpp)
```

Comment:
```suggestion
```
we could remove it in case it's relative path is out-dated

Reply:
As long as the reader looks at the clp version used in this commit it can't be out-dated. It is also important for future readers to know the origin in case there were meaningful changes applied to the source.

Reply:
My concern is this comment can be outdated if we update CLP dependency version. When you update the underlying go ffi implementation, it is easy to miss we have a comment on the Go side pointing to C++ code (could happen where both the author and the reviewer are not aware of this comment).

Reply:
It is a good point that the advantage of keeping the comment likely doesn't outweigh the potential confusion.